notebooks/Visualizing Geo Data.ipynb

```diff
@@ -0,0 +1,320 @@
+    "#open a previously prepared file containing average long term rainfall data\n",
+    "#the csv file was prepared from TRMM3B42ALT_199801_201505.tif\n",
+    "#availbe from: ftp://africagrids.net/30000m/TRMM3B42/Average/\n",
+    "df = pd.read_csv('../data/mapping_data/avg_rainfall.csv')\n",
+    "df.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "import matplotlib.cm as cm\n",
+    "import matplotlib\n",
+    "from matplotlib.colors import LogNorm\n",
+    "\n",
+    "def convert_to_hex(rgba_color):\n",
+    "    \"\"\"A hex converter\"\"\"\n",
+    "    #note that currently this only uses rgb. \n",
+    "    #the transparency informatoin in 'a' is discarded\n",
+    "    red = int(rgba_color[0]*255)\n",
+    "    green = int(rgba_color[1]*255)\n",
+    "    blue = int(rgba_color[2]*255)\n",
+    "    return '#{r:02x}{g:02x}{b:02x}'.format(r=red,g=green,b=blue)\n",
+    "\n",
+    "\n",
+    "def color(x, scalemin=df['avg_rainfall'].min(), scalemax=df['avg_rainfall'].max(), log=True):\n",
+    "    \"\"\"This function assigns a color based on the value of x\"\"\"\n",
+    "    if log:\n",
+    "        norm=LogNorm(vmin=scalemin, vmax=scalemax)\n",
+    "    else:\n",
+    "        norm =matplotlib.colors.Normalize(vmin=scalemin, vmax=scalemax)\n",
+    "    #select a color map from matplotlib\n",
+    "    cmap = cm.RdYlGn\n",
+    "    #get a scalar map based on the normalization and color map we choose\n",
+    "    m = cm.ScalarMappable(norm=norm, cmap=cmap)\n",
+    "    #return the rbga color that the value of x corresponds to\n",
+    "    #convert it to hex for plotting with folium\n",
+    "    return convert_to_hex(m.to_rgba(x))\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "af = folium.Map(location=[-4, 21], zoom_start=3, tiles='cartodbpositron')\n",
+    "\n",
+    "#Foilium requires a list of [lat, long] points, not an array\n",
+    "plist = []\n",
+    "for lat, long in zip(df.lat.values,df.long.values):\n",
+    "    plist.append([lat,long])\n",
+    "\n",
+    "#since there are so many points we need to downsample\n",
+    "#otherwise we won't be able to plot\n",
+    "downsample = 100\n",
+    "for idx, p in enumerate(plist[::downsample]):\n",
+    "    rain = df.avg_rainfall.values[idx*downsample]\n",
+    "    col_idx = color(rain, log=False)\n",
+    "    folium.CircleMarker(location=p,radius=2, popup=\"%0.1f inches\" %rain, fill_color=col_idx, color=col_idx).add_to(af)\n",
+    "\n",
+    "af"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introducing the RUNNING_LOCALLY variable\n",
+    "\n",
+    "Binder currently can't import geopandas, but you can download this notebook and run it locally after installing the\n",
```

Comment:
I'd phrase this as "There are [known issues](https://forums.databricks.com/questions/10096/libgeos-not-found-needed-for-shapely.html) installing some of `geopandas`' dependencies on Linux that preclude us from running some of our mapping code remotely. If you'd like to run this code yourself, you'll need to clone the repo to your own machine and install the dependencies noted in [requirements.txt](../requirements.txt)"